### PR TITLE
feat(go): add support for bumping a Go version file

### PR DESCRIPTION
**Pull request by codyoss**

### CODE REVIEW

**Review by bcoe — APPROVED** [review]

File: test/releasers/go-yoshi.ts [hunk 1]
```diff
@@ -67,10 +72,13 @@ describe('GoYoshi', () => {
         releasePR.gh,
         'getFileContentsOnBranch'
       );
-      // CHANGELOG is not found, and will be created:
       getFileContentsStub
-        .onCall(0)
-        .rejects(Object.assign(Error('not found'), {status: 404}));
+        .withArgs('version.go', 'master')
+        .resolves(buildFileContent('version.go'));
```
Comment (bcoe) [inline test/releasers/go-yoshi.ts:77]: I think it might be worth adding a test case that confirms that a missing `version.go` is a `noop`.

**Review by bcoe — CHANGES_REQUESTED** [review]
Small nit requesting an additional test case. I apologize for the slow review, feel free to assign a PR to someone explicitly if this ever happens.

File: test/releasers/go-yoshi.ts (hunk 1 above)
Comment (codyoss) [reply test/releasers/go-yoshi.ts:77]: I can add an explicit test if you think that is needed but I showed in a couple of followup commits this is confirmed by current tests
Screen Shot 2021-11-15 at 2 13 52 PM
:

**Comment by bcoe** [comment]
@codyoss we have this huge refactor which I think is pretty much ready to land:

https://github.com/chingor13/rp3

@chingor13 should we land this, and merge it into your refactor. Or land your refactor and merge this work into it?

**Comment by bcoe** [comment]
@codyoss okay, refactor is complete, sorry for doing that to you.

**Comment by chingor13** [comment]
Reimplemented in #1112 1112

**Pull request closed without merging**